pennylane/templates/layers/random.py

```diff
@@ -230,6 +232,13 @@ def compute_decomposition(
          CNOT(wires=['b', 'a']),
          RX(tensor(1.4000), wires=['a'])]
         """
+        if ratio_imprimitive:
```

Comment:
if this is set, and all previous values are defined positionally, won't they all be off-by-one and wrong?

Reply:
If they defined it positionally, then its now `ratio_imprim` and no warning is raised.

```
qml.RandomLayers.compute_decomposition(weights,["a", "b"], 0.3,
                                    imprimitive=qml.CNOT, rotations=rotations, seed=42)
```
will raise no warnings.

Reply:
I guess if they start with `(weights,["a", "b"], ratio_imprimitive=0.3, ...` then python forces you to give names to all other kwargs so there should be no problem